compiled/man/man4/app.4
.TH app 5 "kernel 2.16.3" "Ericsson AB" "Files"
.SH NAME
app \- Application resource file.
.SH DESCRIPTION
.LP
The \fIapplication resource file\fR\& specifies the resources an application uses, and how the application is started\&. There must always be one application resource file called \fIApplication\&.app\fR\& for each application \fIApplication\fR\& in the system\&.
.LP
The file is read by the application controller when an application is loaded/started\&. It is also used by the functions in \fIsystools\fR\&, for example when generating start scripts\&.
.SH "FILE SYNTAX"

.LP
The application resource file should be called \fIApplication\&.app\fR\& where \fIApplication\fR\& is the name of the application\&. The file should be located in the \fIebin\fR\& directory for the application\&.
.LP
It must contain one single Erlang term, which is called an \fIapplication specification\fR\&:
.LP
.nf

{application, Application,
  [{description,  Description},
   {id,           Id},
   {vsn,          Vsn},
   {modules,      Modules},
   {maxP,         MaxP},
   {maxT,         MaxT},
   {registered,   Names},
   {included_applications, Apps},
   {applications, Apps},
   {env,          Env},
   {mod,          Start},
   {start_phases, Phases}]}.

             Value                Default
             -----                -------
Application  atom()               -
Description  string()             ""
Id           string()             ""
Vsn          string()             ""
Modules      [Module]             []
MaxP         int()                infinity
MaxT         int()                infinity
Names        [Name]               []
Apps         [App]                []
Env          [{Par,Val}]          []
Start        {Module,StartArgs}   []
Phases       [{Phase,PhaseArgs}]  undefined
  Module = Name = App = Par = Phase = atom()
  Val = StartArgs = PhaseArgs = term()
.fi
.LP
\fIApplication\fR\& is the name of the application\&.
.LP
For the application controller, all keys are optional\&. The respective default values are used for any omitted keys\&.
.LP
The functions in \fIsystools\fR\& require more information\&. If they are used, the following keys are mandatory: \fIdescription\fR\&, \fIvsn\fR\&, \fImodules\fR\&, \fIregistered\fR\& and \fIapplications\fR\&\&. The other keys are ignored by \fIsystools\fR\&\&.
.RS 2
.TP 2
.B
\fIdescription\fR\&:
A one-line description of the application\&.
.TP 2
.B
\fIid\fR\&:
Product identification, or similar\&.
.TP 2
.B
\fIvsn\fR\&:
The version of the application\&.
.TP 2
.B
\fImodules\fR\&:
All modules introduced by this application\&. \fIsystools\fR\& uses this list when generating start scripts and tar files\&. A module can only be defined in one application\&.
.TP 2
.B
\fImaxP\fR\&:
\fIDeprecated - will be ignored\fR\& 
.br
The maximum number of processes allowed in the application\&.
.TP 2
.B
\fImaxT\fR\&:
The maximum time in milliseconds that the application is allowed to run\&. After the specified time the application will automatically terminate\&.
.TP 2
.B
\fIregistered\fR\&:
All names of registered processes started in this application\&. \fIsystools\fR\& uses this list to detect name clashes between different applications\&.
.TP 2
.B
\fIincluded_applications\fR\&:
All applications which are included by this application\&. When this application is started, all included application will automatically be loaded, but not started, by the application controller\&. It is assumed that the topmost supervisor of the included application is started by a supervisor of this application\&.
.TP 2
.B
\fIapplications\fR\&:
All applications which must be started before this application is allowed to be started\&. \fIsystools\fR\& uses this list to generate correct start scripts\&. Defaults to the empty list, but note that all applications have dependencies to (at least) \fIkernel\fR\& and \fIstdlib\fR\&\&.
.TP 2
.B
\fIenv\fR\&:
Configuration parameters used by the application\&. The value of a configuration parameter is retrieved by calling \fIapplication:get_env/1,2\fR\&\&. The values in the application resource file can be overridden by values in a configuration file (see \fIconfig(4)\fR\&) or by command line flags (see \fIerl(1)\fR\&)\&.
.TP 2
.B
\fImod\fR\&:
Specifies the application callback module and a start argument, see \fIapplication(3)\fR\&\&.
.RS 2
.LP
The \fImod\fR\& key is necessary for an application implemented as a supervision tree, or the application controller will not know how to start it\&. The \fImod\fR\& key can be omitted for applications without processes, typically code libraries such as the application STDLIB\&.
.RE
.TP 2
.B
\fIstart_phases\fR\&:
A list of start phases and corresponding start arguments for the application\&. If this key is present, the application master will - in addition to the usual call to \fIModule:start/2\fR\& - also call \fIModule:start_phase(Phase,Type,PhaseArgs)\fR\& for each start phase defined by the \fIstart_phases\fR\& key, and only after this extended start procedure will \fIapplication:start(Application)\fR\& return\&.
.RS 2
.LP
Start phases may be used to synchronize startup of an application and its included applications\&. In this case, the \fImod\fR\& key must be specified as:
.RE
.LP
.nf

{mod, {application_starter,[Module,StartArgs]}}
.fi
.RS 2
.LP
The application master will then call \fIModule:start/2\fR\& for the primary application, followed by calls to \fIModule:start_phase/3\fR\& for each start phase (as defined for the primary application) both for the primary application and for each of its included application, for which the start phase is defined\&.
.RE
.RS 2
.LP
This implies that for an included application, the set of start phases must be a subset of the set of phases defined for the primary application\&. Refer to \fIOTP Design Principles\fR\& for more information\&.
.RE
.RE
.SH "SEE ALSO"

.LP
\fBapplication(3)\fR\&, systools(3)

changes.txt
Dateien:

cstt.view.ui.loginWindow
cstt.view.loginWindow
cstt.view.ui.csttViewport
cstt.store.resellerStore
cstt.store.yearStore

app.js

ldap_auth_gateway.erl
ldap_auth.erl
rewrites.json

local.ini

authentication_handlers = {couch_httpd_auth, default_authentication_handler}, {ldap_auth, handle_admin_role}

[ldap_auth]
AuthenticationHandlers = {couch_httpd_auth, cookie_authentication_handler}, {ldap_auth, handle_basic_auth_req}

curl -vX POST http://cstt3.cordaware.com:5990/_session -H 'Content-Type: application/x-www-form-urlencoded' -d 'name=michael.scharrer&password=CW$feb2014'


yearStore laden!
-----------
Benutzer Dokument enthält
disabled: true
{"error":"unauthorized","reason":"User is disabled for login"}
-----------
wenn Benutzer nicht als Dokument in der DB cstt existiert
{"error":"unauthorized","reason":"User is not allowed to login"}



Todo:

_changes feed für Datenbank cstt_web


9:02:55 PM
"/Users/michael/bin/Sencha/Cmd/4.0.4.84/sencha" app build testing
cd /Users/michael/Entwicklung/cstt/cstt_web/_attachments
/Users/michaelscharrer/bin/Sencha/Cmd/4.0.4.84/sencha app build production

/Users/michaelscharrer/bin/Sencha/Cmd/4.0.4.84/sencha app refresh


cstt_web5/_attachments/app.js
// Alle Klassen einbinden
Ext.require('*');

Ext.application({ 
    name: 'cstt',
    appFolder: 'app',
    controllers: [
        'customer',
        'tickets',
        'csttController',
        'authController'
    ],
    autoCreateViewport: true,
    launch: function() {
    	
    	Ext.tip.QuickTipManager.init();

delete Ext.tip.Tip.prototype.minWidth;

// Default Timeout auf 5 min setzen
Ext.Ajax.timeout = 5*60*1000; //5 minutes


Ext.util.ms_print = function(element) {

	 var myWindow = window.open('', '', 'width=800,height=600');
	        myWindow.document.write('<html><head>');
	        myWindow.document.write('<title>' + 'Drucken' + '</title>');
	        myWindow.document.write('<link rel="Stylesheet" type="text/css" href="css/print.css" />');
	        myWindow.document.write('</head><body>');
	        myWindow.document.write(element.body.dom.innerHTML);
	        myWindow.document.write('</body></html>');
	        myWindow.print();
}

          // Add the additional 'advanced' VType
    Ext.apply(Ext.form.field.VTypes, {
                password: function(val, field) {
            if (field.initialPassField) {
                var pwd = field.up('form').down('#' + field.initialPassField);
                return (val == pwd.getValue());
            }
            return true;
        },

        passwordText: 'Passwort stimmen nicht &uuml;berein!'
    }); 
    
    Ext.apply(Ext.form.field.VTypes, {
	uniqueNameText: 'Der angegebene Wert ist bereits vorhanden. Bitte w&auml;hlen Sie einen anderen Wert/Text.',
    //  vtype validation function
    uniqueName: function(val, field) {

    if(!Ext.isEmpty(field.uniqueURL)) {
    	
    	if(typeof field.originalValue != 'undefined' && field.isEqual(Ext.util.Format.lowercase(val), Ext.util.Format.lowercase(field.originalValue))) {
    		return true;	
    	}
    	
		var trimValue = Ext.util.Format.trim(val);			
      
		var lowercase = Ext.util.Format.lowercase(trimValue);
		var uppercase = Ext.util.Format.uppercase(trimValue);
      
		// TODO: Hier müssen noch Anführungszeichen durch Bachslashes gesetzt werden
      
 			Ext.Ajax.request({
                method: 'GET',
                async: false,
                url: field.uniqueURL + '?startkey="'+encodeURIComponent(lowercase)+'"&endkey="'+encodeURIComponent(uppercase)+'"',
                // Header auf application/json setzen - CouchDB
                headers: {
                    'Content-Type': 'application/json; charset=utf-8'
                },
                callback : function(opts, success, response) {
				var json = Ext.JSON.decode(response.responseText);

		                 if (!Ext.isEmpty(json.rows) && json.rows.length > 0) {
							result = false;

		                 } else {
		                 	result = true;
		                 }
                	
   				 }
            });
            
    	}    

     return result;
    }
});



Ext.apply(Ext.form.field.VTypes, {
    tvs : function(val, field) {
	// Nur GIF, PNG, JPEG, JPG erlaubt
        var fileName = /^.*\.(tvs)$/i;
        return fileName.test(val);
    },
    tvsText : "Erlaubte Dateiendungen für Remotesitzungen: (*.tvs)"
 
});


        // Override for adding tooltips to form fields
Ext.override(Ext.form.Field, {
    afterRender : function() 
    {         
        this.callParent(arguments);   
        try {
            if(this.biTooltipText)
            {
                create_tooltip_fields(this, this.biTooltipText);
            }
        } catch(e){}
    }
});

function create_tooltip_fields(field, tooltipText)
{
    Ext.QuickTips.register({
        target:  field.getEl(),
        title: '',
        text: '<span style="">' + tooltipText +'</span>',
        enabled: true,
        trackMouse: true
    });
    var label = findLabel(field);
    if(label)
    {
        Ext.QuickTips.register({
            target:  label,
            title: '',
            text: '<span style="">' + tooltipText +'</span>',
            enabled: true,
            trackMouse: true
        });
    }
}

var findLabel = function(field) {
    
    var wrapDiv = null;
    var label = null

    // find form-element and label
    wrapDiv = field.getEl().up('div.x-form-element');
    if(wrapDiv)
        label = wrapDiv.child('label');
    
    if(label)
        return label;
    
    // find form-item and label
    wrapDiv = field.getEl().up('div.x-form-item');
    if(wrapDiv)
        label = wrapDiv.child('label');
    
    if(label)
        return label;
}


        
        
        
        Ext.define('Ext.form.action.SubmitCouchDB', {
    		override: 'Ext.form.action.Submit',
    		onSuccess: function(response) {

				var form = this.form,
				success = true,
				result = this.processResponse(response);
				
				// Änderungen für CouchDB Dateiuploads
				if(form.owner.couchUpload === true) {
					result = true;
					
				}
				// Ende Änderungen für CouchDB Dateiuploads
				
				if (result !== true && !result.success) {
				if (result.errors) {
				form.markInvalid(result.errors);
				}
				this.failureType = Ext.form.action.Action.SERVER_INVALID;
				success = false;
				}
				
				form.afterAction(this, success);
			}

    	});
    	

    	
        // Benutzerdefinierter VType f&uuml;r Versionsverwaltung
        Ext.apply(Ext.form.VTypes, {
            versionBI:  function(v) {
                return /^([0-9]).([0-9]).([0-9]).([0-9])$/.test(v);
            },
            versionBIText: 'Erlaubtes Format f&uuml;r Version, z. B.: 5.1.0.3',
            versionBIMask: /[\d\.]/i
        });	
    	
    	Ext.example = function(){
            var msgCt;

            function createBox(t, s){
                // return ['<div class="msg">',
                //         '<div class="x-box-tl"><div class="x-box-tr"><div class="x-box-tc"></div></div></div>',
                //         '<div class="x-box-ml"><div class="x-box-mr"><div class="x-box-mc"><h3>', t, '</h3>', s, '</div></div></div>',
                //         '<div class="x-box-bl"><div class="x-box-br"><div class="x-box-bc"></div></div></div>',
                //         '</div>'].join('');
                return '<div class="msg"><h3>' + t + '</h3><p>' + s + '</p></div>';
            }
            return {
                msg : function(title, format){
                    if(!msgCt){
                        msgCt = Ext.core.DomHelper.insertFirst(document.body, {id:'msg-div'}, true);
                    }
                    var s = Ext.String.format.apply(String, Array.prototype.slice.call(arguments, 1));
                    var m = Ext.core.DomHelper.append(msgCt, createBox(title, s), true);
                    m.hide();
                    m.slideIn('t').ghost("t", { delay: 4000, remove: true});
                },

                init : function(){
                    //            var t = Ext.get('exttheme');
                    //            if(!t){ // run locally?
                    //                return;
                    //            }
                    //            var theme = Cookies.get('exttheme') || 'aero';
                    //            if(theme){
                    //                t.dom.value = theme;
                    //                Ext.getBody().addClass('x-'+theme);
                    //            }
                    //            t.on('change', function(){
                    //                Cookies.set('exttheme', t.getValue());
                    //                setTimeout(function(){
                    //                    window.location.reload();
                    //                }, 250);
                    //            });
                    //
                    //            var lb = Ext.get('lib-bar');
                    //            if(lb){
                    //                lb.show();
                    //            }
                }
            };
        }();




        // Cookie
        Ext.state.Manager.setProvider(new Ext.state.CookieProvider({
            expires: new Date(new Date().getTime()+(1000*60*60*24*30)) // 30 Tage ab jetzigem Zeitpunkt
        }));
        
        // Disable state management for all components by default:
        Ext.Component.prototype.stateful = false;
    	
    }
});
